File: packages/llm-test-harness/src/metrics/custom.ts
import type { MetricFn, MetricScore } from '../types.js'

export class Custom implements MetricFn {
  constructor(
    readonly name: string,
    private fn: (text: string) => MetricScore | Promise<MetricScore>,
  ) {}

  evaluate(text: string): MetricScore | Promise<MetricScore> {
    return this.fn(text)
  }
}

File: packages/llm-test-harness-python/llm_test_harness/metrics/custom.py
from __future__ import annotations

from typing import Callable

from ..types import MetricFn, MetricScore


class Custom(MetricFn):
    def __init__(self, name: str, fn: Callable[[str], MetricScore]) -> None:
        self._name = name
        self._fn = fn

    @property
    def name(self) -> str:
        return self._name

    def evaluate(self, text: str) -> MetricScore:
        return self._fn(text)
